packages/react/src/utils/getChildren.ts

```diff
@@ -40,7 +40,7 @@ export const getChildElementsEvenIfContainerInbetween = (children: React.ReactNo
 
   /* If there's a container element in between, we dig out the child elements from within. */
   if (childrenHasContainer && React.isValidElement(arrayChildren[0])) {
-    return React.Children.toArray(arrayChildren[0].props.children);
+    return getChildrenAsArray(arrayChildren[0].props.children);
   }
-  return arrayChildren;
+  return arrayChildren as React.ReactElement[];
```

Comment:
I think it would be better to have the function's parameters have this type instead of casting.

Reply:
This whole function can be removed when header Children API refactoring is done.